HealthcareSystem/Backend/Repository/PatientRepository/MySqlPatientRepository/MySqlActivePatientRepository.cs

```diff
@@ -44,7 +44,17 @@ public List<Patient> GetAllPatients()
         }
         public Patient GetPatientByJmbg(string jmbg)
         {
-            return _context.Patients.Find(jmbg);
+            try
+            {
+                Patient patient = _context.Patients.Find(jmbg);
+                if (patient == null)
+                    throw new DatabaseException("Patient doesn't exist in database.");
+                return _context.Patients.Find(jmbg);
+            }
+            catch (Exception)
+            {
+                throw new DatabaseException("The database connection is down.");
+            }
```

Comment:
This is something @sladjana-savkovic noticed yesterday, but this implementation has some issues. Specifically, you throw an exception at line 51, but you're inside a try block, and then catch Exception. So the DatabaseException you throw here will get caught and replaced anyway (and I believe it should be a NotFoundException you're throwing instead of DatabaseException, too?).

Reply:
Resolved in f220bf17acde3b181f97f446b33ccf25ff9c819a.